src/routes/login/+page.server.ts

```diff
@@ -0,0 +1,67 @@
+import { lucia } from "$lib/server/auth";
+import { fail, redirect, type Actions } from "@sveltejs/kit";
+import { verify } from "@node-rs/argon2";
+import db from "$lib/db";
+
+export const actions: Actions = {
+	default: async (event) => {
+		const formData = await event.request.formData();
+		const username = formData.get("username");
+		const password = formData.get("password");
+
+		if (
+			typeof username !== "string" ||
+			username.length < 3 ||
+			username.length > 31 ||
+			!/^[a-z0-9_-]+$/.test(username)
+		) {
+			return fail(400, {
+				message: "Invalid username"
+			});
+		}
+		if (typeof password !== "string" || password.length < 6 || password.length > 255) {
+			return fail(400, {
+				message: "Invalid password"
+			});
+		}
+
+		const existingUser = await db.query.users.findFirst({
+			where: (table, { eq }) => {return eq(table.firstname, username)}
```

Comment:
The username cannot be firstname use email instead

Reply:
Good catch!